iob_soc.py

```diff
@@ -33,325 +34,325 @@
 from axi_interconnect import axi_interconnect
 
 
+@dataclass
```

Comment:
We are going to use it to validate user inputs in the core.py file

Reply:
Seems like we are going to use `mypy` https://github.com/IObundle/iob-soc/pull/799
So we no longer need the `@validate_types_class` decorator.

Most of the attributes in the `iob_soc.py` class don't need to define their data types because they are already inherited from iob_module (and their data types are defined there).